src/iam/iam-base.ts

```diff
@@ -181,14 +181,13 @@ export class IAMBase {
     }): Promise<AccountInfo | undefined> {
         const { privateKey, rpcUrl } = this._connectionOptions;
         if (this._executionEnvironment === ExecutionEnvironment.NODE) {
```

Comment:
I think that from now on we don't need to consider execution environment to initialize signer, so you can remove this block

Reply:
But this will make a breaking change, because Node user will need to set WalletProvider while initializingConnection, am I right?
What's more, for `WalletProvider.PrivateKey` i'm setting `this._providerType = walletProvider;`. Can it also be set for Node env?

Reply:
For now we can set providerType can be set in constructor if node environment is detected